### PR TITLE
Added description path to listing page

### LINKED ISSUE #388: Display path on search results view
As a user, I would like to see the context for the learning resources in my search results. 

Ideally, the items in the path would be links to open the learning resource panel for the corresponding container resource. 

May need to add path field (in model) to elastic search index.

### PR DESCRIPTION
**Pull request by giocalitri**
Implements the path as sequence of titles of the learning resources " / " separated.
An important part of this PR is the migration that populates the new field of the learningresource model.

Closes #388 
[#99424610]

### CODE REVIEW

File: learningresources/api.py [hunk 1]
```diff
@@ -301,3 +305,44 @@ def import_static_assets(course, path):
...
                 create_static_asset(course.id, django_file)
+
+
+def join_description_paths(*args):
+    """
+    Helper function to format the description path.
+    Args:
+        list of dpath (unicode): description path
+    Returns:
+        formatted dpath (unicode)
```
Comment (noisecapella) [inline learningresources/api.py]: For the argument you should use the format `args (type here): description path`. Whatever you think is sensible for the type is fine with me

The return section should be `unicode: Formatted dpath`

File: learningresources/management/commands/update_description_paths.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+class Command(BaseCommand):
+    """
+    Command for sync_permissions
+    """
+    help = "Synchronizes and updates the description_path"
+
+    def handle(self, *args, **options):
+        """Command handler"""
+        for learning_resourse in LearningResource.objects.all():
```
Comment (noisecapella) [inline learningresources/management/commands/update_description_paths.py]: Change to `learning_resource`

File: learningresources/api.py [hunk 3]
```diff
@@ -301,3 +305,44 @@ def import_static_assets(course, path):
...
+    """
+    formatted_dpath = join(
+        *[' {0} '.format(dpath) for dpath in args if dpath != '']
+    )
+    return formatted_dpath.strip()
+
+
+def update_description_path(resource, force_parent_update=False):
```
Comment (noisecapella) [inline learningresources/api.py:321]: Would it make sense to move this function to a method in `LearningResource`? The only required argument is the `resource` model, it doesn't return anything and mutates the `LearningResource`

Comment (noisecapella) [inline learningresources/api.py:321]: Oh, it looks like that may interfere with your migration

File: learningresources/tests/test_api.py [hunk 4]
```diff
@@ -190,5 +184,105 @@ def test_create_static_asset(self):
...
+    def test_update_description_path(self):
+        """Tests for update_description_path"""
+        # after created a resource without parent has the description path
+        # equal to the title
+        self.assertIsNone(self.resource.parent)
+        self.assertEqual(
+            self.resource.title,
+            self.resource.description_path
+        )
+        # changing the title does not update the descriptio path automatically
```
Comment (noisecapella) [inline learningresources/tests/test_api.py]: Typo in `descriptio`

File: learningresources/models.py [hunk 5]
```diff
@@ -125,6 +125,7 @@ class LearningResource(BaseModel):
     description = models.TextField(blank=True)
     content_xml = models.TextField()
     materialized_path = models.TextField()
+    description_path = models.TextField(blank=True)
```
Comment (noisecapella) [inline learningresources/models.py:128]: Why are we making this a field and not just creating this value on the fly?

File: learningresources/tests/test_api.py [hunk 6]
```diff
@@ -190,5 +184,105 @@ def test_create_static_asset(self):
...
+        )
+        # create a child resource
+        child_res = self.create_resource(
+            parent=self.resource
+        )
+        # the description path is the combination of the child resource title
+        # and the parent description path
+        self.assertEqual(
+            child_res.description_path,
+            api.join_description_paths(
```
Comment (noisecapella) [inline learningresources/tests/test_api.py:248]: In any of the invocations of `join_description_paths` are we passing in an empty string? If not we should assert that the empty string is skipped

Comment (giocalitri) [inline learningresources/tests/test_api.py:248]: no, we are almost never passing an empty string, but I added tests anyway and I changed the way I join the strings

File: learningresources/api.py [hunk 7]
```diff
@@ -301,3 +305,41 @@ def import_static_assets(course, path):
...
+    """
+    Updates the specified learning resource description path
+    based on the current title and the parent's description path
+    Args:
+        resource (learningresources.models.LearningResource): LearningResource
+        force_parent_update (boolean): force parent update
+    Returns:
+        None
+    """
+    description_path = ''
```
Comment (noisecapella) [inline learningresources/api.py:331]: Since we are setting this in both paths we can remove this line

File: learningresources/tests/base.py [hunk 8]
```diff
@@ -61,17 +66,19 @@ def create_resource(self, **kwargs):
...
+            content_xml=kwargs.get('content_xml', "<blah>other blah</blah>"),
+            mpath=kwargs.get('mpath', "/otherblah"),
+            url_name=kwargs.get('url_name'),
+            dpath=''
         )
         learn_res.xa_nr_views = kwargs.get('xa_nr_views', 0)
         learn_res.xa_nr_attempts = kwargs.get('xa_nr_attempts', 0)
         learn_res.xa_avg_grade = kwargs.get('xa_avg_grade', 0)
         learn_res.save()
+        update_description_path(learn_res)
```
Comment (noisecapella) [inline learningresources/tests/base.py:81]: I think this means we have two saves happening unnecessarily. No strong opinion on this though, it's probably not often called

**Pull request merged**